source/common/upstream/cluster_manager_impl.h

```diff
@@ -763,6 +764,7 @@ class ClusterManagerImpl : public ClusterManager,
     Common::CallbackHandlePtr priority_update_cb_;
     // Keep smaller fields near the end to reduce padding
     const bool added_via_api_ : 1;
+    const bool ignore_removal_ : 1;
```

Comment:
I think that the name `ignore_removal_` is somewhat ambiguous. Specifically, it is unclear what should happen when the `remove_ignored` argument of `removeCluster()` is `true` while `ignore_removal_` is also true (I understand that it is specified in the method's comment, but there may be a way to improve upon this).
Please rename to something like `avoid_cds_removal_`, which specifies that it should only avoid CDS removals.

Reply:
Right! `avoid_cds_removal_` sounds better. 
For `remove_ignored` argument of removeCluster(), how if we rename it `force_remove`?